### PR TITLE
add sign-in route and twitter ajax calls

### PR DESCRIPTION
**Pull request by sbatson5**
# Why

This PR adds a login route for OAuth with twitter.  This PR also adds a session service, so that we can check the state of our logged in user at any time.

Depends on https://github.com/sbatson5/movie_wager_api/pull/8

### CODE REVIEW

File: app/sign-in/route.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import Ember from 'ember';
+
+export default Ember.Route.extend({
+  ajax: Ember.inject.service(),
+
+  actions: {
+    signIn() {
+      this.get('ajax').request('/api/v1/twitter-auth', {
+        method: 'POST',
+        foo: 'bar'
```
Comment (michaelpelletier) [inline app/sign-in/route.js]: Leftover debugging?

File: app/models/user.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import DS from 'ember-data';
+
+const {
+  Model,
+  attr
+} = DS;
+
+export default Model.extend({
+  twitterId: attr(),
```
Comment (michaelpelletier) [inline app/models/user.js]: 🔤

File: app/router.js [hunk 3]
```diff
@@ -8,6 +8,8 @@ const Router = Ember.Router.extend({
 
 Router.map(function() {
   this.route('admin');
+  this.route('sign-in');
+  this.route('authenticated');
```
Comment (michaelpelletier) [inline app/router.js:11]: :abc:

File: app/services/session.js [hunk 4]
```diff
@@ -0,0 +1,59 @@
+import Ember from 'ember';
+
+const {
+  Service,
```
Comment (michaelpelletier) [inline app/services/session.js]: 🔤

File: app/application/route.js [hunk 5]
```diff
@@ -15,8 +15,17 @@ export default Route.extend({
...
-  setupController(controller) {
-    this._setCurrentUser(controller);
+  model() {
+    const session = get(this, 'session');
+    return session.fetchSession().then(() => {
+      return session.getUser();
+    }).catch(() => {
+      return null;
+    });
+  },
```
Comment (scanieso) [inline app/application/route.js:25]: This change looks great to me! 👍

File: app/services/session.js [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+    let store = get(this, 'store');
+    const modelClass = store.modelFor('user');
+    const serializer = store.serializerFor('user');
+    const normalizedPayload = serializer.normalizeSingleResponse(store, modelClass, payload, null, true);
+
+    this._setUser(store.push(normalizedPayload));
+  },
+
+  _deleteUser() {
+    set(this, 'currentUser', null);
```
Comment (RaisedByWolves) [inline app/services/session.js:53]: I'm not familiar enough with Ember - These user-destructive actions are just client-side actions, correct? They are not at all connected to the DB?

**Comment by RaisedByWolves** [comment]
I'm not familiar enough with this style of architecture for web clients, but I rubbed my human eyes on the changes and found nothing that appeared to be accidentally checked in. Either sensitive credentials, commented out partial code or anything like that.
There did seem to be too much in session.js - things about twitter authentication etc that might be better in their own module.

File: app/services/session.js (hunk 6 above)
Comment (sbatson5) [reply app/services/session.js:53]: Right. This just stores state on our friend end for convenience. It is destroyed when the app is destroyed or reloaded anyway.  This is not persisted.

`currentUser` is a property on this object.

**Comment by sbatson5** [comment]
@RaisedByWolves `session.js` shouldn't have any info relating to twitter. It simply hits the API's `session` endpoint and loads the user, or allows the logged in user to destroy their session. The idea is that all session logic (i.e. checking the session, setting the `currentUser` and logging the user out) should live in this service.

**Pull request merged**